File: Plugins/AM_MSXML_Bruker_PlugIn/clsAnalysisResourcesMSXMLBruker.vb
'*********************************************************************************************************
' Written by Matthew Monroe for the US Department of Energy 
' Pacific Northwest National Laboratory, Richland, WA
' Created 03/30/2011
'
'*********************************************************************************************************

Option Strict On

Imports AnalysisManagerBase

Public Class clsAnalysisResourcesMSXMLBruker
    Inherits clsAnalysisResources


#Region "Methods"
    ''' <summary>
	''' Retrieves files necessary for creating the .mzXML file
    ''' </summary>
    ''' <returns>IJobParams.CloseOutType indicating success or failure</returns>
    ''' <remarks></remarks>
    Public Overrides Function GetResources() As IJobParams.CloseOutType

        'Get input data file
        Dim strRawDataType As String = m_jobParams.GetParam("RawDataType")
        Dim eRawDataType = GetRawDataType(strRawDataType)

        Select Case eRawDataType
			Case eRawDataTypeConstants.BrukerFTFolder, eRawDataTypeConstants.BrukerTOFBaf
				' This dataset type is acceptable
			Case Else
				m_message = "Dataset type " & strRawDataType & " is not supported"
				clsLogTools.WriteLog(clsLogTools.LoggerTypes.LogFile, clsLogTools.LogLevels.DEBUG, "clsDtaGenResources.GetResources: " & m_message & "; must be " & RAW_DATA_TYPE_BRUKER_FT_FOLDER & " or " & RAW_DATA_TYPE_BRUKER_TOF_BAF_FOLDER)
				Return IJobParams.CloseOutType.CLOSEOUT_FAILED

		End Select

		If Not RetrieveSpectra(strRawDataType) Then
			clsLogTools.WriteLog(clsLogTools.LoggerTypes.LogFile, clsLogTools.LogLevels.DEBUG, "clsDtaGenResources.GetResources: Error occurred retrieving spectra.")
			Return IJobParams.CloseOutType.CLOSEOUT_FAILED
		End If

        If Not MyBase.ProcessMyEMSLDownloadQueue(m_WorkingDir, MyEMSLReader.Downloader.DownloadFolderLayout.SingleDataset) Then
            Return IJobParams.CloseOutType.CLOSEOUT_FAILED
        End If

        Return IJobParams.CloseOutType.CLOSEOUT_SUCCESS

    End Function
#End Region

End Class

File: Plugins/AM_MSXML_Bruker_PlugIn/clsAnalysisResourcesMSXMLBruker.cs
//*********************************************************************************************************
// Written by Matthew Monroe for the US Department of Energy
// Pacific Northwest National Laboratory, Richland, WA
// Created 03/30/2011
//
//*********************************************************************************************************

using AnalysisManagerBase;

namespace AnalysisManagerMsXmlBrukerPlugIn
{
    /// <summary>
    /// Retrieve resources for the MSXml Bruker plugin
    /// </summary>
    public class clsAnalysisResourcesMSXMLBruker : clsAnalysisResources
    {
        #region "Methods"

        /// <summary>
        /// Retrieves files necessary for creating the .mzXML file
        /// </summary>
        /// <returns>CloseOutType indicating success or failure</returns>
        public override CloseOutType GetResources()
        {
            // Retrieve shared resources, including the JobParameters file from the previous job step
            var result = GetSharedResources();
            if (result != CloseOutType.CLOSEOUT_SUCCESS)
            {
                return result;
            }

            // Get input data file
            var rawDataType = mJobParams.GetParam("RawDataType");
            var eRawDataType = GetRawDataType(rawDataType);

            switch (eRawDataType)
            {
                case eRawDataTypeConstants.BrukerFTFolder:
                case eRawDataTypeConstants.BrukerTOFBaf:
                    break;
                // This dataset type is acceptable
                default:
                    mMessage = "Dataset type " + rawDataType + " is not supported";
                    LogDebug(
                        "clsDtaGenResources.GetResources: " + mMessage + "; must be " + RAW_DATA_TYPE_BRUKER_FT_FOLDER + " or " +
                        RAW_DATA_TYPE_BRUKER_TOF_BAF_FOLDER);

                    return CloseOutType.CLOSEOUT_FAILED;
            }

            if (!FileSearch.RetrieveSpectra(rawDataType))
            {
                LogDebug("clsDtaGenResources.GetResources: Error occurred retrieving spectra.");
                return CloseOutType.CLOSEOUT_FAILED;
            }

            if (!base.ProcessMyEMSLDownloadQueue(mWorkDir, MyEMSLReader.Downloader.DownloadLayout.SingleDataset))
            {
                return CloseOutType.CLOSEOUT_FAILED;
            }

            return CloseOutType.CLOSEOUT_SUCCESS;
        }

        #endregion
    }
}
